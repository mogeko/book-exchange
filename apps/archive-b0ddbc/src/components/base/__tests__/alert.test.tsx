import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Alert from "@/components/base/alert";

describe("Alert", () => {
  it("renders an error alert correctly", () => {
    const { container } = render(<Alert.Error message="Error!" />);

    expect(container.querySelector("div")).toHaveClass("alert alert-error");
    expect(container.querySelector("div")).toHaveTextContent("Error!");
  });

  it("snapshot an error alert", () => {
    const { container } = render(<Alert.Error message="Error!" />);

    expect(container).toMatchSnapshot();
  });

  it("renders a warning alert correctly", () => {
    const { container } = render(<Alert.Warning message="Warning!" />);

    expect(container.querySelector("div")).toHaveClass("alert alert-warning");
    expect(container.querySelector("div")).toHaveTextContent("Warning!");
  });

  it("snapshot a warning alert", () => {
    const { container } = render(<Alert.Warning message="Warning!" />);

    expect(container).toMatchSnapshot();
  });

  it("renders an info alert correctly", () => {
    const { container } = render(<Alert.Info message="Info!" />);

    expect(container.querySelector("div")).toHaveClass("alert alert-info");
    expect(container.querySelector("div")).toHaveTextContent("Info!");
  });

  it("snapshot an info alert", () => {
    const { container } = render(<Alert.Info message="Info!" />);

    expect(container).toMatchSnapshot();
  });

  it("renders a success alert correctly", () => {
    const { container } = render(<Alert.Success message="Success!" />);

    expect(container.querySelector("div")).toHaveClass("alert alert-success");
    expect(container.querySelector("div")).toHaveTextContent("Success!");
  });

  it("snapshot a success alert", () => {
    const { container } = render(<Alert.Success message="Success!" />);

    expect(container).toMatchSnapshot();
  });
});
