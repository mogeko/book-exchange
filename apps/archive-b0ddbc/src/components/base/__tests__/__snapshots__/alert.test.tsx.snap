// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert snapshot a success alert 1`] = `
<div>
  <div
    class="alert alert-success shadow-lg"
  >
    <div>
      <svg
        class="w-6 h-6"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.27 10.87h.71l4.56-4.56-.71-.71-4.2 4.21-1.92-1.92L4 8.6l2.27 2.27z"
        />
        <path
          clip-rule="evenodd"
          d="M8.6 1c1.6.1 3.1.9 4.2 2 1.3 1.4 2 3.1 2 5.1 0 1.6-.6 3.1-1.6 4.4-1 1.2-2.4 2.1-4 2.4-1.6.3-3.2.1-4.6-.7-1.4-.8-2.5-2-3.1-3.5C.9 9.2.8 7.5 1.3 6c.5-1.6 1.4-2.9 2.8-3.8C5.4 1.3 7 .9 8.6 1zm.5 12.9c1.3-.3 2.5-1 3.4-2.1.8-1.1 1.3-2.4 1.2-3.8 0-1.6-.6-3.2-1.7-4.3-1-1-2.2-1.6-3.6-1.7-1.3-.1-2.7.2-3.8 1-1.1.8-1.9 1.9-2.3 3.3-.4 1.3-.4 2.7.2 4 .6 1.3 1.5 2.3 2.7 3 1.2.7 2.6.9 3.9.6z"
          fill-rule="evenodd"
        />
      </svg>
      <span>
        Success!
      </span>
    </div>
  </div>
</div>
`;

exports[`Alert snapshot a warning alert 1`] = `
<div>
  <div
    class="alert alert-warning shadow-lg"
  >
    <div>
      <svg
        class="w-6 h-6"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M7.56 1h.88l6.54 12.26-.44.74H1.44L1 13.26 7.56 1zM8 2.28L2.28 13H13.7L8 2.28zM8.625 12v-1h-1.25v1h1.25zm-1.25-2V6h1.25v4h-1.25z"
          fill-rule="evenodd"
        />
      </svg>
      <span>
        Warning!
      </span>
    </div>
  </div>
</div>
`;

exports[`Alert snapshot an error alert 1`] = `
<div>
  <div
    class="alert alert-error shadow-lg"
  >
    <div>
      <svg
        class="w-6 h-6"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8.6 1c1.6.1 3.1.9 4.2 2 1.3 1.4 2 3.1 2 5.1 0 1.6-.6 3.1-1.6 4.4-1 1.2-2.4 2.1-4 2.4-1.6.3-3.2.1-4.6-.7-1.4-.8-2.5-2-3.1-3.5C.9 9.2.8 7.5 1.3 6c.5-1.6 1.4-2.9 2.8-3.8C5.4 1.3 7 .9 8.6 1zm.5 12.9c1.3-.3 2.5-1 3.4-2.1.8-1.1 1.3-2.4 1.2-3.8 0-1.6-.6-3.2-1.7-4.3-1-1-2.2-1.6-3.6-1.7-1.3-.1-2.7.2-3.8 1-1.1.8-1.9 1.9-2.3 3.3-.4 1.3-.4 2.7.2 4 .6 1.3 1.5 2.3 2.7 3 1.2.7 2.6.9 3.9.6zM7.9 7.5L10.3 5l.7.7-2.4 2.5 2.4 2.5-.7.7-2.4-2.5-2.4 2.5-.7-.7 2.4-2.5-2.4-2.5.7-.7 2.4 2.5z"
          fill-rule="evenodd"
        />
      </svg>
      <span>
        Error!
      </span>
    </div>
  </div>
</div>
`;

exports[`Alert snapshot an info alert 1`] = `
<div>
  <div
    class="alert alert-info shadow-lg"
  >
    <div>
      <svg
        class="w-6 h-6"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8.568 1.031A6.8 6.8 0 0 1 12.76 3.05a7.06 7.06 0 0 1 .46 9.39 6.85 6.85 0 0 1-8.58 1.74 7 7 0 0 1-3.12-3.5 7.12 7.12 0 0 1-.23-4.71 7 7 0 0 1 2.77-3.79 6.8 6.8 0 0 1 4.508-1.149zM9.04 13.88a5.89 5.89 0 0 0 3.41-2.07 6.07 6.07 0 0 0-.4-8.06 5.82 5.82 0 0 0-7.43-.74 6.06 6.06 0 0 0 .5 10.29 5.81 5.81 0 0 0 3.92.58zM7.375 6h1.25V5h-1.25v1zm1.25 1v4h-1.25V7h1.25z"
          fill-rule="evenodd"
        />
      </svg>
      <span>
        Info!
      </span>
    </div>
  </div>
</div>
`;
